src/Parsers/Access/ParserGrantQuery.cpp

```diff
@@ -284,8 +283,49 @@ bool ParserGrantQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     AccessRightsElements elements;
     std::shared_ptr<ASTRolesOrUsersSet> roles;
-    if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
-        return false;
+
+    bool current_grants = false;
+    if (!is_revoke && ParserKeyword{"CURRENT GRANTS"}.ignore(pos, expected))
+    {
+        current_grants = true;
+        if (ParserToken(TokenType::OpeningRoundBracket).ignore(pos, expected))
+        {
+            if (!parseElementsWithoutOptions(pos, expected, elements))
+                return false;
+
+            if (!ParserToken(TokenType::ClosingRoundBracket).ignore(pos, expected))
+                return false;
+
+            /// If no elements were specified it will grant all available for user grants.
+            /// Using `.size() == 0` because `.empty()` is overridden and returns true for NONE elements.
+            /// Specifically, this should handle `GRANT CURRENT GRANTS() ...`
+            if (elements.size() == 0) // NOLINT
+                elements.emplace_back(AccessType::ALL);
```

Comment:
I think we don't need this syntax with nothing between parentheses, it's not obvious and the functionality can be easily written in other ways.